help.py

```diff
@@ -41,3 +42,276 @@ def datetime_to_epoch_millis(date: datetime) -> int:
+
+        for k,v in args.__dict__.items():
+            if k in list(survey_payload.keys()):
+                survey_payload[k] = v if v != None else survey_payload.pop(k)
+
+        if args.iocfile and args.ioctype is None:
+            sys.exit("--iocfile requires --ioctype")
+
+        if args.iocfile and not os.path.isfile(args.ioc_file):
+            sys.exit('Supplied --iocfile is not a file')
+
+        if (args.output or args.prefix) and args.no_file:
+            sys.exit('--output and --prefix cannot be used with --no-file')
+
+        if args.days and args.minutes:
+            sys.exit('--days and --minutes are mutually exclusive')
+
+        if (args.sigmarule or args.sigmadir) and edr == 'cortex':
+            sys.exit('Neither --sigmarule nor --sigmadir are supported by product "cortex"')
+
+        if (args.sigmarule or args.sigmadir) and edr == 's1' and not args.dv:
+            sys.exit('Neither --sigmarule nor --sigmadir are supported by SentinelOne PowerQuery')
+
+        if args.sigmarule and not os.path.isfile(args.sigmarule):
+            sys.exit('Supplied --sigmarule is not a file')
+
+        if args.sigmadir and not os.path.isdir(args.sigmadir):
+            sys.exit('Supplied --sigmadir is not a directory')
+
+        log = logger(edr=args.edr,logs_dir=args.log_dir)
+
+        definition_files = list()
+        definitions = dict()
+        sigma_rules = list()
+
+        # test if deffile exists
+        # deffile can be resolved from 'definitions' folder without needing to specify path or extension
+        if args.deffile:
+            if not os.path.exists(args.deffile):
+                repo_deffile: str = os.path.join(os.path.dirname(__file__), 'definitions', args.deffile)
+                if not repo_deffile.endswith('.json'):
+                    repo_deffile = repo_deffile + '.json'
+
+                if os.path.isfile(repo_deffile):
+                    log.debug(f'Using repo definition file {repo_deffile}')
+                    args.deffile = repo_deffile
+                else:
+                    sys.exit("The deffile doesn't exist. Please try again.")
+            definition_files.append(args.deffile)
+
+        # if --defdir add all files to list
+        if args.defdir:
+            if not os.path.exists(args.defdir):
+                sys.exit("The defdir doesn't exist. Please try again.")
+            else:
+                for root_dir, dirs, files in os.walk(args.defdir):
+                    for filename in files:
+                        if os.path.splitext(filename)[1] == '.json':
+                            definition_files.append(os.path.join(root_dir, filename))
+
+        if definitions: survey_payload['definitions'] = definitions
```

Comment:
I don't see where `definitions : dict` is ever defined except on line 65 of surveyor.py where it's set to be a `list`?